src/networks.ts

```diff
@@ -4,74 +4,26 @@ import { networks } from "bitcoinjs-lib";
  * This module exports network constants and provide some utility
  * functions for displaying the network name and passing the network
  * value to bitcoinjs.
- *
- * @module networks
  */
-
-/**
- * Constant corresponding to the Bitcoin mainnet.
- *
- * @constant
- * @type {string}
- * @default mainnet
- *
- */
-export const MAINNET = "mainnet";
-
-/**
- * Constant corresponding to the current Bitcoin testnet.
- *
- * @constant
- * @type {string}
- * @default testnet
- */
-export const TESTNET = "testnet";
-
-/**
- * Constant corresponding to a bitcoin regtest network
- *
- * @constant
- * @type {string}
- * @default regtest
- */
-export const REGTEST = "regtest";
-
-/**
- * Constant corresponding to a bitcoin signet network
- *
- * @constant
- * @type {string}
- * @default signet
- */
-export const SIGNET = "signet";
-
-/**
- * Enumeration of possible values for bitcoin networks ([MAINET]{@link module:networks.MAINNET}|[TESTNET]{@link module:networks.TESTNET}).
- *
- * @constant
- * @enum {string}
- * @default
- */
-export const NETWORKS = {
-  MAINNET,
-  TESTNET,
-  REGTEST,
-} as const;
+/* eslint-disable no-shadow */
+export enum Network {
+  MAINNET = "mainnet",
+  TESTNET = "testnet",
+  REGTEST = "regtest",
+  SIGNET = "signet",
+}
```

Comment:
Could we not keep the individual exports as well? It would be nice to still be able to do `import { MAINNET, TESTNET } from "unchained-bitcoin" ` So we could probably update to this: 

```typescript
export const MAINNET = "mainnet";

export enum Network {
  MAINNET = MAINNET,
  // ... etc.
}
```

Reply:
I think keeping the individual exports will complexify the typing. If an enum value is expected, then providing a string constant will not be acceptable. There was also a comment that was removed suggesting that the constants could be removed after enums were built.

Reply:
What's the real goal here, a flexible API or a strict API? Enum ads strictness with a helpful way to reference. Union literals would be the most strict without helpful reference. If the goal is to make it the most strict without making the type confusing, I'd argue that the best would be a union literal that is not exported from the lib. Lib consumers would be required to pass a string matching one of the values specified by docs/IDE. This would probably have the least confusing typing.

Reply:
Yeah, all fair. I think was just thinking if there would be a way to minimize the diff. But this is definitely cleaner overall I think.